v3/src/execution/__tests__/execution.test.ts

```diff
@@ -0,0 +1,366 @@
+  T
+][];
+
+const validQueries: JestInput<{ query: string; operationName?: string; variables?: VariableValues }> = [
+  [
+    "basic - { people { name } }",
+    {
+      query: gql`
+      query Basic {
+        people {
+          name
+        }
+      }
+    `
+    }
+  ],
+  [
+    "with nested selection sets - { people { name friends { name friends { name } } } }",
+    {
+      query: gql`
+      query Nested {
+        people {
+          name
+          friends {
+            name
+            friends {
+              name
+            }
+          }
+        }
+      }
+    `
+    }
+  ],
+  [
+    "with directives - { people { name { friends { name } } }",
+    {
+      query: gql`
+      query Directives($includeFriends: Boolean!) {
+        people {
+          name
+          friends @include(if: $includeFriends) {
+            name
+          }
+        }
+      }
+      `,
+      variables: { includeFriends: true }
+    }
+  ],
+  [
+    "with multiple documents - { people { name } }",
+    {
+      query: gql`
+      query FirstDocument {
+        people {
+          name
+        }
+      }
+
+      query SecondDocument {
+        people {
+          id
+        }
+      }
+    `,
+      operationName: "FirstDocument"
+    }
+  ]
+];
+
+const unparseableQueries: JestInput<string>= [
+  [
+    "missing closing }",
+    // Using gql on this unparseable document breaks code highlighting everywhere
+    `
+      query Basic {
+        people {
+          name
+        }
+    `
+  ],
+  [
+    "empty selection set",
+    gql`
+      query Nested {
+        people { }
+      }
+    `
+  ],
+  [
+    "empty query",
+    gql`
+      query Empty {}
+    `
+  ]
+];
+
+const invalidQueries: JestInput<{ query: string; operationName?: string; }> = [
+  [
+    "basic",
+    {
+      query: gql`
+      query Basic {
+        invalidField
+      }
+    `
+    }
+  ],
+  [
+    "with invalid selection sets",
+    {
+      query: gql`
+      query Nested {
+        people {
+          name
+          invalidField
+        }
+      }
+    `
+    }
+  ],
+  [
+    "missing directive",
+    {
+      query: gql`
+      query MissingDirective {
+        people @invalid {
+          name
+        }
+      }
+      `
+    }
+  ],
+  [
+    "with multiple errors",
+    {
+      query: gql`
+      query FirstDocument {
+        people {
+          invalidOne
+        }
+        invalidTwo
+      }
+    `
+    }
+  ]
+];
+
+const nonExecutableQueries: JestInput<{ query: string }> = [
+  ['error field throws', {
+    query: gql`
+      query {
+        error
+      }
+    `
+  }],
+  ['throws multiple errors', {
+    query: gql`
+      query {
+        error
+        people {
+          nestedError
+        }
+      }
+    `
+  }]
+]
+
+const serverTeam = [
+  { id: 1, name: "James Baxley" },
+  { id: 2, name: "Ashi Krishnan" },
+  { id: 3, name: "Jesse Rosenberger" },
+  { id: 4, name: "Trevor Scheer" }
+];
+
+const schema = makeExecutableSchema({
```

Comment:
Do we have an alternative to `makeExecutableSchema` that we want to use? This introduces `graphql-tools` to our `devDependencies`.

Reply:
Resolved in: 69a571c